packages/near-membrane-base/src/blue.ts

```diff
@@ -34,56 +35,42 @@ import {
     RedConstructor,
     RedFunction,
     RedValue,
+    BlueArrayOrObject,
+    RedArrayOrObject,
 } from './types';
 
-function renameFunction(provider: RedFunction, receiver: BlueFunction) {
-    try {
-        // a revoked proxy will break the membrane when reading the function name
-        const nameDescriptor = ReflectGetOwnPropertyDescriptor(provider, 'name')!;
-        ReflectDefineProperty(receiver, 'name', nameDescriptor);
-    } catch {
-        // intentionally swallowing the error because this method is just extracting the
-        // function in a way that it should always succeed except for the cases in which
-        // the provider is a proxy that is either revoked or has some logic to prevent
-        // reading the name property descriptor.
-    }
-}
-
 const ProxyCtor = Proxy;
 const { isArray: isArrayOrNotOrThrowForRevoked } = Array;
 
-function createBlueShadowTarget(target: BlueProxyTarget): BlueShadowTarget {
+// eslint-disable-next-line no-shadow
+enum TargetTraits {
+    IsArray = 1 << 0,
+    IsFunction = 1 << 1,
+    IsObject = 1 << 2,
+    IsArrowFunction = 1 << 3,
+}
+
+function createShadowTarget(
+    targetTraits: TargetTraits,
```

Comment:
I'm not even sure if this is correct typing... knowing that TargetTraits is enum of bitwise masks.

Reply:
this is still an open question though... is this the right way to type the bitmasks?

Reply:
My understanding of the TS docs says yes.